src/components/contestdetails/ContestDetails.tsx

```diff
@@ -107,34 +106,48 @@ export default function ContestDetails(props: ContestDetailsProps) {
     )
   }
 
-  function showPhotoDrawer(): boolean {
-    // Don't show the ContestSubmission component when the contest
-    // is open because that is handled by OpenContestDetails
-    return contest?.status !== "open"
+  function showContestViewer(): boolean {
+    return contest?.status === "closed"
   }
 
-  function showContestSubmission(submission: Submission): boolean {
-    // Don't show the ContestSubmission component when the contest
-    // is open because that is handled by OpenContestDetails
-    return contest?.status !== "open"
+  function showPhotoDrawer(): boolean {
+    return contest?.status === "closed"
   }
 
-  function showImageInViewer(submission: Submission) {
-    return contest?.status !== "open" || submission.id === auth.currentUser!.uid
+  function showContestSubmission(submission: Submission): boolean {
```

Comment:
Since all of these are looking for "closed" we can probably just add a case in the switch statement below instead

Reply:
yup I'll have an upcoming PR where I add `ClosedContestDetails` so bear with this ugliness for just a while!